test/__snapshots__/bindings.test.ts.snap

```diff
@@ -241,17 +516,131 @@ public interface Name extends software.amazon.jsii.JsiiSerializable {
 
             if (!first.equals(that.first)) return false;
             if (!last.equals(that.last)) return false;
-            return this.middle != null ? this.middle.equals(that.middle) : that.middle == null;
+            if (this.middle != null ? !this.middle.equals(that.middle) : that.middle != null) return false;
+            if (this.objectWithPropertiesAndBooleanAdditionalProperties != null ? !this.objectWithPropertiesAndBooleanAdditionalProperties.equals(that.objectWithPropertiesAndBooleanAdditionalProperties) : that.objectWithPropertiesAndBooleanAdditionalProperties != null) return false;
+            if (this.objectWithPropertiesAndTypedAdditionalProperties != null ? !this.objectWithPropertiesAndTypedAdditionalProperties.equals(that.objectWithPropertiesAndTypedAdditionalProperties) : that.objectWithPropertiesAndTypedAdditionalProperties != null) return false;
+            return this.objectWithTypedAdditionalPropertiesOnly != null ? this.objectWithTypedAdditionalPropertiesOnly.equals(that.objectWithTypedAdditionalPropertiesOnly) : that.objectWithTypedAdditionalPropertiesOnly == null;
         }
 
         @Override
         public final int hashCode() {
             int result = this.first.hashCode();
             result = 31 * result + (this.last.hashCode());
             result = 31 * result + (this.middle != null ? this.middle.hashCode() : 0);
+            result = 31 * result + (this.objectWithPropertiesAndBooleanAdditionalProperties != null ? this.objectWithPropertiesAndBooleanAdditionalProperties.hashCode() : 0);
+            result = 31 * result + (this.objectWithPropertiesAndTypedAdditionalProperties != null ? this.objectWithPropertiesAndTypedAdditionalProperties.hashCode() : 0);
+            result = 31 * result + (this.objectWithTypedAdditionalPropertiesOnly != null ? this.objectWithTypedAdditionalPropertiesOnly.hashCode() : 0);
             return result;
         }
     }
 }
 "
 `;
+
+exports[`language bindings 4`] = `
+"// 1bc04b5291c26a46d918139138b992d2de976d6851d0893b0476b85bfbdfc6e6
+package myorg
+
+
+type Name struct {
+	First *string \`json:\\"first\\"\`
+	Last *string \`json:\\"last\\"\`
+	Middle *string \`json:\\"middle\\"\`
+	ObjectWithPropertiesAndBooleanAdditionalProperties *NameObjectWithPropertiesAndBooleanAdditionalProperties \`json:\\"objectWithPropertiesAndBooleanAdditionalProperties\\"\`
+	ObjectWithPropertiesAndTypedAdditionalProperties *NameObjectWithPropertiesAndTypedAdditionalProperties \`json:\\"objectWithPropertiesAndTypedAdditionalProperties\\"\`
+	ObjectWithTypedAdditionalPropertiesOnly *map[string]*string \`json:\\"objectWithTypedAdditionalPropertiesOnly\\"\`
+}
+
+type NameObjectWithPropertiesAndBooleanAdditionalProperties struct {
+	Name *string \`json:\\"name\\"\`
+}
```

Comment:
@eladb , I would expect here to contain `AdditionalProperties map[string]interface{}`